src/class_sqlThread.py

```diff
@@ -29,22 +18,120 @@
+    from . import tr
+    from .bmconfigparser import BMConfigParser
+    from .debug import logger
+    from .addresses import encodeAddress
 
+# pylint: disable=attribute-defined-outside-init,protected-access
+root_path = os.path.dirname(os.path.dirname(__file__))
 
-class sqlThread(threading.Thread):
+
+def connection_build():
+    """
+        Stablish SQL connection
+    """
+    conn = sqlite3.connect(state.appdata + 'messages.dat')
+    conn.text_factory = str
+    cur = conn.cursor()
+    return conn, cur
+
+
+class UpgradeDB(object):
+    """
+        Upgrade Db with respect to versions
+    """
+
+    conn, cur = connection_build()
+
+    def __init__(self):
+        self.current_level = None
+        self.max_level = 11
+
+    def __get_current_settings_version(self):
+        """
+            Upgrade Db with respect to their versions
+        """
+        item = '''SELECT value FROM settings WHERE key='version';'''
+        parameters = ()
+        self.cur.execute(item, parameters)
+        return int(self.cur.fetchall()[0][0])
+
+    def _upgrade_one_level_method(self, level):
+        """
+            Apply switcher to call methods accordingly
+        """
+
+        if level != self.__get_current_settings_version():
+            return None
+
+        # Migrate Db with level
+        method_name = 'upgrade_schema_data_' + str(level)
+        method = getattr(self, method_name, lambda: "Invalid version")
+        return method()
+
+    def _upgrade_one_level_sql_statement(self, file_name):
+        """
+            Execute SQL files and queries
+        """
+        try:
+            with open(os.path.join(root_path, "src/sql/init_version_{}.sql".format(file_name))) as sql_file:
+                sql_as_string = sql_file.read()
+            self.cur.executescript(sql_as_string)
+        except Exception as err:
+            if str(err) == 'table inbox already exists':
+                return "table inbox already exists"
+            else:
+                sys.stderr.write(
+                    'ERROR trying to create database file (message.dat). Error message: %s\n' % str(err))
+                os._exit(0)
+
+    def upgrade_to_latest(self, cur, conn):
+        """
+            Initialise upgrade level
+        """
+
+        # Declare variables
+        self.conn = conn
+        self.cur = cur
+        self.current_level = self.__get_current_settings_version()
+        self.max_level = 11
+
+        # call upgrading level in loop
+        for l in range(self.current_level, self.max_level):
+            if int(l) == 3:
+                continue
+            self._upgrade_one_level_method(l)
+            self._upgrade_one_level_sql_statement(l)
+
+    def increment_settings_version(self, level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (level + 1,)
+        self.cur.execute(item, parameters)
+
```

Comment:
this should be more transparent. For example, you can use a `@property` decorator for properties `sql_schema_version` and `configparser_schema_version` and use these instead calling the methods directly. And setter can enforce a `+1` irrespective of what you pass as an argument.